### PR TITLE
update k8s client library and fix teleport issue

### PR DESCRIPTION
**Pull request by Orzelius**
commits have more details

### CODE REVIEW

File: core/src/plugins/kubernetes/api.ts [hunk 1]
```diff
@@ -88,23 +88,21 @@ const requestAgent = new Agent({ lookup })
 // NOTE: be warned, the API of the client library is very likely to change
 
 type K8sApi =
-  | ApiextensionsV1beta1Api
+  | ApiextensionsV1Api
   | AppsV1Api
   | CoreApi
   | CoreV1Api
-  | ExtensionsV1beta1Api
```
Comment (Orzelius) [inline core/src/plugins/kubernetes/api.ts:95]: This seemed not to be used anywhere and doesn't have a corresponding type in the new client version

**Review by edvald — APPROVED** [review]

**Comment by Orzelius** [comment]
tests are not happy, I'm looking into it

**Comment by Orzelius** [comment]
I had some trouble with the new logging logic in the client but I've got it right down. Good to review!

**Review by edvald — APPROVED** [review]

**Comment by Orzelius** [comment]
I'm rebasing as there's a lot of info in the commits and I've squashed the fixups

**Comment by highb** [comment]
Thank you! I'll test this out against our dev cluster when I have a few spare cycles.

**Pull request merged**